src/stateManager/StateManager.ts

```diff
@@ -0,0 +1,778 @@
+                    "StateManager - onBlockConfirmation - Internal Error - flag undefined"
+                );
+            }
+
+            await this.processConfirmationDecision(
+                signedBlock,
+                confirmationSignature as SignatureLike,
+                finalExecutionFlag
+            );
+        }
+    }
+
+    //Aplies a transaction to the state machine and returns the encoded state with a success callback
+    public async applyTransaction(transaction: TransactionStruct): Promise<{
+        success: boolean;
+        encodedState: string;
+        successCallback: () => void;
+    }> {
+        let { success, successCallback } =
+            await this.stateMachine.stateTransition(transaction);
+        return {
+            success,
+            encodedState: await this.stateMachine.getState(),
+            successCallback
+        };
+    }
+
+    // Used when authoring a block - Executes the transaction and returns a signed block
+    public async playTransaction(
+        tx: TransactionStruct
+    ): Promise<SignedBlockStruct> {
+        await this.mutex.lock();
+
+        try {
+            console.log("Play Transaction", this.getForkCnt());
+            this.ensureChannelIsOpen();
+            await this.ensureItIsMyTurn();
+            this.adjustTimestampIfNeeded(tx);
+
+            const { previousStateHash, encodedState, successCallback } =
+                await this.applyTransactionOrThrow(tx);
+
+            const block = await this.createBlock(tx, previousStateHash);
+            const signedBlock = await this.signBlock(block);
+
+            this.agreementManager.addBlock(
+                block,
+                signedBlock.signature as SignatureLike,
+                encodedState
+            );
+
+            successCallback();
+            await this.onSuccessCommon();
+
+            this.scheduleTask(
+                () => this.maybePostBlockOnChain(block, signedBlock),
+                this.timeConfig.agreementTime * 1000,
+                "maybePostBlockOnChain"
+            );
+
+            return signedBlock;
+        } finally {
+            this.mutex.unlock();
+        }
+    }
+
+    private async maybePostBlockOnChain(
+        block: BlockStruct,
+        signedBlock: SignedBlockStruct
+    ): Promise<void> {
+        // If not everyone has signed, do the on-chain post
+        if (!this.agreementManager.didEveryoneSignBlock(block)) {
+            console.log("Posting calldata on chain!");
+            this.p2pEventHooks.onPostingCalldata?.();
+            this.stateChannelManagerContract
+                .postBlockCalldata(signedBlock)
+                .then((txResponse) => txResponse.wait())
+                .catch((error) => {
+                    console.log("Error posting block on chain", error);
+                });
+        }
+    }
+
+    // returns participants who haven't signed the block
+    public async getPlayersWhoHaventSignedBlock(
```

Comment:
I think the naming should be generic as SDK isn't tied to gaming

Reply:
Yes - players->participants

Reply:
done